toponetx/classes/combinatorial_complex.py

```diff
@@ -447,9 +448,256 @@ def add_cell(self, cell, rank=None, **attr):
+        weight=None,
+        sparse: bool = True,
+        index: bool = False,
+    ):
+        """Compute incidence matrix.
+
+        An incidence matrix indexed by r-ranked hyperedges k-ranked hyperedges
+        r !=k, when k is None incidence_type will be considered instead
+
+        Parameters
+        ----------
+        incidence_type : {'up', 'down'}, default='up'
+        sparse : bool, default=True
+        index : bool, default=False
+            If True return will include a dictionary of children uid : row number
+            and element uid : column number
+
+        Returns
+        -------
+        incidence_matrix : scipy.sparse.csr.csr_matrix or np.ndarray
+        row dictionary : dict
+            Dictionary identifying row with item in entityset's children
+        column dictionary : dict
+            Dictionary identifying column with item in entityset's uidset
+
+        Notes
+        -----
+        Incidence_matrix method  is a method for generating the incidence matrix of a ColoredHyperGraph.
+        An incidence matrix is a matrix that describes the relationships between the hyperedges
+        of a complex. In this case, the incidence_matrix method generates a matrix where
+        the rows correspond to the hyperedges of the complex and the columns correspond to the faces
+        . The entries in the matrix are either 0 or 1,
+        depending on whether a hyperedge contains a given face or not.
+        For example, if hyperedge i contains face j, then the entry in the ith
+        row and jth column of the matrix will be 1, otherwise it will be 0.
+
+        To generate the incidence matrix, the incidence_matrix method first creates
+        a dictionary where the keys are the faces of the complex and the values are
+        the hyperedges that contain that face. This allows the method to quickly look up
+        which hyperedges contain a given face. The method then iterates over the hyperedges in
+        the HyperEdgeView instance, and for each hyperedge, it checks which faces it contains.
+        For each face that the hyperedge contains, the method increments the corresponding entry
+        in the matrix. Finally, the method returns the completed incidence matrix.
         """
-        return super().add_cell(cell, rank, **attr)
+        if rank == to_rank:
+            raise ValueError("incidence must be computed for k!=r, got equal r and k.")
+        if to_rank is None:
+            if incidence_type == "up":
+                children = self.skeleton(rank)
+                uidset = self.skeleton(rank + 1)
+            elif incidence_type == "down":
+                uidset = self.skeleton(rank)
+                children = self.skeleton(rank - 1)
+            else:
+                raise TopoNetXError("incidence_type must be 'up' or 'down' ")
+        else:
+            assert (
+                rank != to_rank
+            )  # incidence is defined between two skeletons of different ranks
```

Comment:
This can be removed. You already checked that case at the beginning of the function.